code/modules/mob/living/basic/basic_mob.dm

```diff
@@ -0,0 +1,257 @@
+	var/friendly_verb_continuous = "nuzzles"
+	/// Attacking, but without damage, verb in present simple tense.
+	var/friendly_verb_simple = "nuzzle"
+
+	// Attack related vars.
+	// Unclear how much of this should either be componentized or pulled to /mob/living
+	var/attacked_sound = 'sound/weapons/punch1.ogg'
+	/// The amount of damage done to the mob when hand-attacked on harm intent.
+	var/harm_intent_damage = 3
+	/// 1 for full damage, 0 for none, -1 for 1:1 heal from that source.
+	var/list/damage_coeff = list(BRUTE = 1, BURN = 1, TOX = 1, STAMINA = 0, OXY = 1)
+	/// Minimum force required to deal any damage
+	var/force_threshold = 0
+
+/mob/living/basic/Initialize(mapload)
+	. = ..()
+
+	if(!loc)
+		stack_trace("Basic mob being instantiated in nullspace")
+
+	apply_atmos_requirements()
+	apply_temperature_requirements()
+
+/mob/living/basic/movement_delay()
+	. = speed
+	if(forced_look)
+		. += DIRECTION_LOCK_SLOWDOWN
+	. += GLOB.configuration.movement.animal_delay
+
+/mob/living/basic/proc/apply_atmos_requirements()
+	if(unsuitable_atmos_damage == 0)
+		return
+
+	atmos_requirements = string_assoc_list(atmos_requirements)
+	AddElement(/datum/element/atmos_requirements, atmos_requirements, unsuitable_atmos_damage)
+
+/mob/living/basic/proc/apply_temperature_requirements()
+	AddElement(/datum/element/body_temperature, minimum_survivable_temperature, maximum_survivable_temperature, unsuitable_cold_damage, unsuitable_heat_damage)
+
+/mob/living/basic/vv_edit_var(vname, vval)
+	switch(vname)
+		if("atmos_requirements", "unsuitable_atmos_damage")
+			RemoveElement(/datum/element/atmos_requirements, atmos_requirements, unsuitable_atmos_damage)
+			. = TRUE
+		if("minimum_survivable_temperature", "maximum_survivable_temperature", "unsuitable_cold_damage", "unsuitable_heat_damage")
+			RemoveElement(/datum/element/body_temperature, minimum_survivable_temperature, maximum_survivable_temperature, unsuitable_cold_damage, unsuitable_heat_damage)
+			. = TRUE
+
+	. = ..()
+
+	switch(vname)
+		if("habitable_atmos", "unsuitable_atmos_damage")
+			apply_atmos_requirements()
+		if("minimum_survivable_temperature", "maximum_survivable_temperature", "unsuitable_cold_damage", "unsuitable_heat_damage")
+			apply_temperature_requirements()
+
+/mob/living/basic/proc/valid_respawn_target_for(mob/user)
+	return FALSE
+
+/mob/living/basic/handle_environment(datum/gas_mixture/readonly_environment)
+	SEND_SIGNAL(src, COMSIG_SIMPLEANIMAL_HANDLE_ENVIRONMENT, readonly_environment)
+
+/mob/living/basic/update_stat(reason = "none given")
+	if(status_flags & GODMODE)
+		return
+	if(stat != DEAD)
+		if(health <= 0)
+			death()
+			create_debug_log("died of damage, trigger reason: [reason]")
+		else
+			if(HAS_TRAIT(src, TRAIT_KNOCKEDOUT))
+				if(stat == CONSCIOUS)
+					KnockOut()
+					create_debug_log("knocked out, trigger reason: [reason]")
+			else if(stat == UNCONSCIOUS)
+				WakeUp()
+				create_debug_log("woke up, trigger reason: [reason]")
```

Comment:
Is this going to cause them to wake up the immediately following tick if they go to sleep?

Reply:
this is the same code as in /mob/living/simple_animal/update_stat, but i don't think we actually ever knock out simplemobs so maybe this has never actually run or been tested.